Revised security guide (#2223)
cap-js/docs: guides/security/overview.md

Instead, **CAP seamlessly integrates with battle-tested [platform services](#btp-services)** that handle these critical security topics centrally. 
This approach not only simplifies the implementation but also enhances security by leveraging robust, well-tested mechanisms provided by the platform.
Built on platform services, CAP allows developers to focus on core application functionality without worrying about the intricacies of security implementation.

Most notably, authentication is covered by CAP-integration of [platform's identity services](./authentication#ias-auth). 
Likewise, TLS termination is offered by the [platform infrastructure](#platform-environment).

![Overview Platform Integration with CAP](./assets/security-platform-integration.drawio.svg){width="600px" }

### Decoupled from Business Logic  { #key-concept-decoupled-coding }

As security functions are factorized into independent components, **application code is entirely decoupled** and hence is not subject to change in case of any security-related adaptions. 

> This safeguards business logic being independent from platform services which are frequently subject to security hardening initiatives.

Reviewer: ```suggestion
This ensures that business logic remains independent of platform services, which are often subject to security-hardening initiatives.
```
